markdown_it/tree.py

```diff
@@ -44,6 +45,12 @@ def __init__(self) -> None:
         # children (i.e. inline or img)
         self.children: List["SyntaxTreeNode"] = []
 
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self.type})"
```

Comment:
Dumb nitpick, but I would use `type(self)` for consistency since we already use that elsewhere in this class (also same reasons we use `str(self)` instead of `self.__str__()` etc.)
```suggestion
        return f"{type(self).__name__}({self.type})"
```